### PR TITLE
Updated guava dependency throughout azure-cosmos

### PR DESCRIPTION
**Pull request by kushagraThapar**
**Motivation**
This PR removes the dependency on the `guava` library due to possibility of recurring vulnerabilities (see [CVE-2020-8908](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2020-8908)) and upgrades the dependency on `testng` library. As a result of moving away from the `guava` library and vulnerabilities in the `commons-codec` library (see [CVE-2020-15250](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2020-15250)) the murmur hash 3 implementation is shaded from the `commons-codec` library.

Furthermore, the apache commons-codec implementation of murmur hash 3 is only used in unit testing and not in any implementation of azure-cosmos SDK.

### CODE REVIEW

**Review by xinlian12 — APPROVED** [review]
LGTM, thanks

**Review by FabianMeiswinkel — APPROVED** [review]
LGTM

File: NOTICE.txt [hunk 1]
```diff
@@ -519,6 +519,9 @@ License notice for Apache Commons Codec
 Copyright 2002-2019 The Apache Software Foundation
 
 commons-codec (https://github.com/apache/commons-codec)
+commons-text (https://github.com/apache/commons-text)
```
Comment (kushagraThapar) [inline NOTICE.txt:522]: Should we update the title to -> 
`License notice for Apache Commons` instead of `License notice for Apache Commons Codec` ?

Comment (jeet1995) [reply NOTICE.txt:522]: Good catch, we should.

**Review by anuchandy — APPROVED** [review]

**Pull request merged**